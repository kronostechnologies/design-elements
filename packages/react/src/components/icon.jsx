import React from 'react';

export default function() {
    return (
        <span></span>
    );
}
