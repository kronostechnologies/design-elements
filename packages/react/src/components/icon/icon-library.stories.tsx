import React from 'react';
import { Icon } from './icon';

export default {
    title: 'Icon/library',
    component: Icon,
};

export const all = () => (
    <>
        <Icon name="alertTriangle" />
        <Icon name="arrowLeft" />
        <Icon name="check" />
        <Icon name="chevronDown" />
        <Icon name="chevronUp" />
        <Icon name="copy" />
        <Icon name="edit" />
        <Icon name="helpCircle" />
        <Icon name="home" />
        <Icon name="info" />
        <Icon name="mail" />
        <Icon name="mapPin" />
        <Icon name="menu" />
        <Icon name="moreHorizontal" />
        <Icon name="open" />
        <Icon name="phone" />
        <Icon name="search" />
        <Icon name="trash" />
        <Icon name="x" />
    </>
);

all.story = {
    decorators: [(storyFn: () => React.ReactElement) => (
        <div style={{ display: 'flex', justifyContent: 'space-around' }}>
            {storyFn()}
        </div>
    )],
};

export const alertTriangle = () => (
    <Icon name="alertTriangle" />
);

export const arrowLeft = () => (
    <Icon name="arrowLeft" />
);

export const check = () => (
    <Icon name="check" />
);

export const chevronDown = () => (
    <Icon name="chevronDown" />
);

export const chevronUp = () => (
    <Icon name="chevronUp" />
);

export const copy = () => (
    <Icon name="copy" />
);

export const edit = () => (
    <Icon name="edit" />
);

export const helpCircle = () => (
    <Icon name="helpCircle" />
);

export const home = () => (
    <Icon name="home" />
);

export const info = () => (
    <Icon name="info" />
);

export const mail = () => (
    <Icon name="mail" />
);

export const mapPin = () => (
    <Icon name="mapPin" />
);

export const menu = () => (
    <Icon name="menu" />
);

export const moreHorizontal = () => (
    <Icon name="moreHorizontal" />
);

export const open = () => (
    <Icon name="open" />
);

export const phone = () => (
    <Icon name="phone" />
);

export const search = () => (
    <Icon name="search" />
);

export const trash = () => (
    <Icon name="trash" />
);

export const x = () => (
    <Icon name="x" />
);
