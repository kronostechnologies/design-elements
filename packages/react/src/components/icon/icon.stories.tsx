import React from 'react';
import { Icon } from './icon';

export default {
    title: 'Icon',
    component: Icon,
};

export const icon = () => (
    <Icon name="alertTriangle" />
);

export const withColor = () => (
    <Icon name="alertTriangle" color="orange" />
);

export const withSize = () => (
    <Icon name="alertTriangle" size="78" />
);
