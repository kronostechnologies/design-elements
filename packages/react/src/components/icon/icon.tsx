import AlertOctagon from 'feather-icons/dist/icons/alert-octagon.svg';
import AlertTriangle from 'feather-icons/dist/icons/alert-triangle.svg';
import ArrowDown from 'feather-icons/dist/icons/arrow-down.svg';
import ArrowLeft from 'feather-icons/dist/icons/arrow-left.svg';
import ArrowRight from 'feather-icons/dist/icons/arrow-right.svg';
import ArrowUp from 'feather-icons/dist/icons/arrow-up.svg';
import Calendar from 'feather-icons/dist/icons/calendar.svg';
import Check from 'feather-icons/dist/icons/check.svg';
import ChevronDown from 'feather-icons/dist/icons/chevron-down.svg';
import ChevronLeft from 'feather-icons/dist/icons/chevron-left.svg';
import ChevronRight from 'feather-icons/dist/icons/chevron-right.svg';
import ChevronUp from 'feather-icons/dist/icons/chevron-up.svg';
import ChevronsLeft from 'feather-icons/dist/icons/chevrons-left.svg';
import ChevronsRight from 'feather-icons/dist/icons/chevrons-right.svg';
import Copy from 'feather-icons/dist/icons/copy.svg';
import Edit from 'feather-icons/dist/icons/edit-2.svg';
import HelpCircle from 'feather-icons/dist/icons/help-circle.svg';
import Home from 'feather-icons/dist/icons/home.svg';
import Info from 'feather-icons/dist/icons/info.svg';
import Mail from 'feather-icons/dist/icons/mail.svg';
import MapPin from 'feather-icons/dist/icons/map-pin.svg';
import Menu from 'feather-icons/dist/icons/menu.svg';
import MoreHorizontal from 'feather-icons/dist/icons/more-horizontal.svg';
import MoreVertical from 'feather-icons/dist/icons/more-vertical.svg';
import Phone from 'feather-icons/dist/icons/phone.svg';
import PlusSign from 'feather-icons/dist/icons/plus.svg';
import Search from 'feather-icons/dist/icons/search.svg';
import Star from 'feather-icons/dist/icons/star.svg';
import Trash from 'feather-icons/dist/icons/trash.svg';
import X from 'feather-icons/dist/icons/x.svg';
import React, { VoidFunctionComponent } from 'react';
import Open from '../../icons/open.svg';
import Reorder from '../../icons/reorder.svg';

const iconMapping = {
    alertTriangle: AlertTriangle,
    alertOctagon: AlertOctagon,
    arrowDown: ArrowDown,
    arrowLeft: ArrowLeft,
    arrowRight: ArrowRight,
    arrowUp: ArrowUp,
    calendar: Calendar,
    check: Check,
    chevronDown: ChevronDown,
    chevronLeft: ChevronLeft,
    chevronRight: ChevronRight,
    chevronUp: ChevronUp,
    chevronsLeft: ChevronsLeft,
    chevronsRight: ChevronsRight,
    copy: Copy,
    edit: Edit,
    helpCircle: HelpCircle,
    home: Home,
    info: Info,
    mail: Mail,
    mapPin: MapPin,
    menu: Menu,
    moreHorizontal: MoreHorizontal,
    moreVertical: MoreVertical,
    open: Open,
    phone: Phone,
    plusSign: PlusSign,
    reorder: Reorder,
    search: Search,
    star: Star,
    trash: Trash,
    x: X,
} as const;

export type IconName = keyof typeof iconMapping;

interface IconProps {
    /** Name of the icon, has to be in IconName */
    name: IconName;
    /**
     * Size will affect both width and height
     * @default 24
     */
    size?: string;
    /**
     * Icon's color, in any format supported by css (keyword, rgb, hex, etc.)
     * @default currentColor
     */
    color?: string;
}

export const Icon: VoidFunctionComponent<IconProps> = ({
    name,
    size,
    color,
    ...props
}: IconProps) => {
    const Component = iconMapping[name];

    if (!Component) {
        return null;
    }

    return (
        <Component
            height={size}
            focusable={false}
            width={size}
            color={color}
            {...props /* eslint-disable-line react/jsx-props-no-spreading */}
        />
    );
};

Icon.defaultProps = {
    size: '24',
    color: 'currentColor',
};
