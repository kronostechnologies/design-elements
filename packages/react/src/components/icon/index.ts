export { Icon } from './icon';
export type { IconName, IconProps } from './icon';
