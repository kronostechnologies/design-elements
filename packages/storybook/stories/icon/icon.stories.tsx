import { Icon } from '@equisoft/design-elements-react';
import React from 'react';

export default {
    title: 'Icon',
    component: Icon,
};

export const icon = () => (
    <Icon name="alertTriangle" />
);

export const withColor = () => (
    <Icon name="alertTriangle" color="orange" />
);

export const withSize = () => (
    <Icon name="alertTriangle" size="78" />
);
