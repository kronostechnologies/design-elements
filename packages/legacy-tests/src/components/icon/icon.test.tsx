import { render } from 'enzyme';
import { Icon } from '~/components/icon/icon';

describe('Icon', () => {
    it('Matches the snapshot', () => {
        const tree = render(<Icon name="home" />);

        expect(tree).toMatchSnapshot();
    });
});
