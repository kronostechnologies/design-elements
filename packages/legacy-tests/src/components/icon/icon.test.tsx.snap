// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Icon Matches the snapshot 1`] = `
<svg
  color="currentColor"
  focusable="false"
  height="24"
  width="24"
/>
`;
